typescript/packages/subsurface-viewer/src/layers/grid3d/privateGrid3dLayer.ts

```diff
@@ -263,12 +264,16 @@ export default class PrivateLayer extends Layer<PrivateLayerProps> {
         parameters:
             | typeof DEFAULT_TEXTURE_PARAMETERS
             | typeof DISCRETE_TEXTURE_PARAMETERS;
+        colorLookupTolerance: number;
     } {
         if (this.props.colorMapFunction instanceof Uint8Array) {
+            const count = this.props.colorMapFunction.length / 3;
             return {
                 imageData: this.props.colorMapFunction,
-                count: this.props.colorMapFunction.length / 3,
+                count,
                 parameters: DISCRETE_TEXTURE_PARAMETERS,
+                //As the colors are not interpolated a slight offset in the texture is needed to avoid "color fighting" when a color is picked on the border between two colors.
+                colorLookupTolerance: (1.0 / count) * 0.5,
```

Comment:
Can count be 0 (if no/empty colormap is sent) ?

Reply:
This case is fixed.